doozer/doozerlib/cli/scan_osh.py

```diff
@@ -1,29 +1,186 @@
+                 check_triggered: Optional[bool], all_builds: Optional[bool], create_jira_tickets: Optional[bool]):
         self.runtime = runtime
         self.last_brew_event = last_brew_event
         self.dry_run = dry_run
         self.brew_tags = []
         self.specific_nvrs = nvrs
         self.check_triggered = check_triggered
         self.all_builds = all_builds
+        self.create_jira_tickets = create_jira_tickets
         self.error_nvrs = []
+        self.version = self.runtime.group.split("-")[-1]
+        self.jira_project = "OCPBUGS"
 
         # Initialize runtime and brewhub session
         self.runtime.initialize(clone_distgits=False)
         self.koji_session = koji.ClientSession(self.runtime.group_config.urls.brewhub)
 
+        # Initialize JIRA client
+        if self.create_jira_tickets:
+            self.jira_client: JIRA = self.runtime.build_jira_client()
+
+    def get_latest_scan_task_id(self, brew_package_name: str):
+        """
+        Find the last successful scan for a particular package and retrieve its ID
+        """
+        # Eg: osh-cli find-tasks --regex ose-network-tools-container-v4.15
+        cmd = f"osh-cli find-tasks --regex {brew_package_name}"
+
+        _, result, _ = cmd_gather(cmd)
+
+        finished_tasks = result.strip().split("\n")
+
+        for task_id in finished_tasks:
+            cmd = f"osh-cli task-info {task_id}"
+            self.runtime.logger.info(f"Running: {cmd}")
+
+            _, result, _ = cmd_gather(cmd)
+
+            # Check if the task finished successfully
+            if "state_label = CLOSED" in result:
+                return task_id
+
+    @staticmethod
+    def check_if_vulnerabilities_exist(task_id: str, nvr: str) -> bool:
+        """
+        ProdSec only wants us to check the scan-results-imp.js file of a scan and see if vulnerabilities are reported.
+        """
+
+        response = requests.get(SCAN_RESULTS_URL_TEMPLATE.format(task_id=task_id, nvr=nvr))
+
+        if len(response.json()["defects"]) > 0:
+            return True
+
+        return False
+
+    def create_update_ocpbugs_ticket(self, packages: dict):
+        """
+        Check if an OCPBUGS ticket already exists for a particular package. If it doesn't create one and update
+        the contents to point to the latest scan results
+        """
+        for brew_package_name in packages:
+            data = packages[brew_package_name]
+
+            # Check if ticket exits
+            summary = f"{self.version} SAST scan issues for {brew_package_name}"
+            query = f"project={self.jira_project} AND ( summary ~ '{summary}' ) AND " \
+                    "statusCategory in ('To Do', 'In Progress')"
+            description = "Vulnerabilities have been detected for this package. Check here for the last scan results: " \
+                          f"{SCAN_RESULTS_URL_TEMPLATE.format(task_id=data['latest_coverity_scan'], nvr=data['nvr'])}"
```

Comment:
Is this an HTML view we can link to as well?

Reply:
Yes, if we remove the `raw` suffix, we get a web view. Added in https://github.com/openshift-eng/art-tools/pull/141/commits/117f1efb6b41a9e4bcd3027f152bda54d83def23